plugin/manager/manager.go

```diff
@@ -0,0 +1,269 @@
+	"path"
+	"path/filepath"
+	"runtime"
+
+	"github.com/notaryproject/notation-go/plugin"
+)
+
+// Plugin represents a potential plugin with all it's metadata.
+type Plugin struct {
+	plugin.Metadata
+
+	Path string `json:",omitempty"`
+
+	// Err is non-nil if the plugin failed one of the candidate tests.
+	Err error `json:",omitempty"`
+}
+
+// ErrNotFound is returned by Manager.Get and Manager.Run when the plugin is not found.
+var ErrNotFound = errors.New("plugin not found")
+
+// ErrNotCompliant is returned by Manager.Run when the plugin is found but not compliant.
+var ErrNotCompliant = errors.New("plugin not compliant")
+
+// commander is defined for mocking purposes.
+type commander interface {
+	// Output runs the command, passing req to the its stdin.
+	// It only returns an error if the binary can't be executed.
+	// Returns stdout if success is true, stderr if success is false.
+	Output(ctx context.Context, path string, command string, req []byte) (out []byte, success bool, err error)
+}
+
+// execCommander implements the commander interface using exec.Command().
+type execCommander struct{}
+
+func (c execCommander) Output(ctx context.Context, name string, command string, req []byte) ([]byte, bool, error) {
+	var stdout, stderr bytes.Buffer
+	cmd := exec.CommandContext(ctx, name, command)
+	cmd.Stdin = bytes.NewReader(req)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	err := cmd.Run()
+	if _, ok := err.(*exec.ExitError); err != nil && !ok {
+		return nil, false, err
+	}
+	if !cmd.ProcessState.Success() {
+		return stderr.Bytes(), false, nil
+	}
+	return stdout.Bytes(), true, nil
+}
+
+// rootedFS is io.FS implementation used in NewManager.
+// root is the root of the file system tree passed to os.DirFS.
+type rootedFS struct {
+	fs.FS
+	root string
+}
+
+// Manager manages plugins installed on the system.
+type Manager struct {
+	fsys  fs.FS
+	cmder commander
+}
+
+// NewManager returns a new manager.
+func NewManager() *Manager {
+	configDir, err := os.UserConfigDir()
+	if err != nil {
+		// Lets panic for now.
+		// Once the config is moved to notation-go we will move this code to
+		// the config package as a global initialization.
+		panic(err)
+	}
+	pluginDir := filepath.Join(configDir, "notation", "plugins")
+	return &Manager{rootedFS{os.DirFS(pluginDir), pluginDir}, execCommander{}}
+}
+
+// Get returns a plugin on the system by its name.
+//
+// If the plugin is not found, the error is of type ErrNotFound.
+// The plugin might be incomplete if p.Err is not nil.
+func (mgr *Manager) Get(ctx context.Context, name string) (*Plugin, error) {
+	return mgr.newPlugin(ctx, name)
+}
+
+// List produces a list of the plugins available on the system.
+//
+// Some plugins might be incomplete if their Err is not nil.
+func (mgr *Manager) List(ctx context.Context) ([]*Plugin, error) {
+	var plugins []*Plugin
+	fs.WalkDir(mgr.fsys, ".", func(dir string, d fs.DirEntry, _ error) error {
+		if dir == "." || !d.IsDir() {
+			return nil
+		}
```

Comment:
We should skip over plugin dirs that are symlinks, something that was missed in the spec.

Reply:
Done!